src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -456,6 +457,14 @@ private function resolveVariables($value)
                 $value = (string) getenv($name);
             }
 
+            if ('' === $value && isset($matches['default_value'])) {
+                if (false !== strpbrk($matches['default_value'], '\'"{$')) {
```

Comment:
Can't we prevent this from happening by already excluding these characters in the regex?

Reply:
I thought that for users that expect their default value to contain one of these characters, it would be nicer to get a clear error message instead of having to figure out the internals of this function to see which characters are excluded.